### PR TITLE
Add limit support for load_dygraph loading jit.save result

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

APIs
### Describe

Add limit support for `load_dygraph` loading `jit.save` result, only supports loading default save format of `jit.save`.

In order to support other compatible formats, we may need to discuss adding new argument to load_dygraph, and this work will be discussed later.

### CODE REVIEW

File: python/paddle/fluid/dygraph/checkpoint.py [hunk 1]
```diff
@@ -140,22 +141,84 @@ def load_dygraph(model_path, keep_name_table=False):
...
+        # 2. load `__variables.info__`
+        var_info_path = os.path.join(model_prefix, EXTRA_VAR_INFO_FILENAME)
+        if not os.path.exists(var_info_path):
+            raise RuntimeError(
+                "No target can be loaded. Now only supports loading `state_dict` from "
+                "the result saved by `imperative.save` and `imperative.jit.save`."
+            )
+        with open(var_info_path, 'rb') as f:
+            extra_var_info = pickle.load(f) if six.PY2 else pickle.load(
                 f, encoding='latin1')
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/checkpoint.py]: Could you add a simple comment about using encoding='latin1'?

I didn't know it before and I don't think it is a common knowledge. I found from python doc that it is required when we unpickle NumPy arrays. Is it the reason?

https://docs.python.org/3/library/pickle.html#pickle.Unpickler

**Review by zhhsplendid — COMMENTED** [review]
Other things LGTM

File: python/paddle/fluid/dygraph/checkpoint.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/checkpoint.py]: thx, I think this is the reason, but here it is not Numpy array, but a dict, I try to remove this limit.

Comment (chenwhql) [reply python/paddle/fluid/dygraph/checkpoint.py]: done, thx

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]

**Pull request merged**